experiments/AMIP/moist_mpi_earth/coupler_driver.jl

```diff
@@ -196,63 +209,84 @@ function solve_coupler!(cs, energy_check)
             SIC = interpolate_midmonth_to_daily(cs.dates.date[1], cs.mode.SIC_info)
             ice_mask = ice_sim.integrator.p.Ya.ice_mask .= get_ice_mask.(SIC, FT)
 
+            # accumulate data at each timestep
+            accumulate_diags(atmos_sim.integrator.u.c, cs.monthly_state_diags)
+
+            # save and reset monthly averages
+            @calendar_callback :(
```

Comment:
Macro hygiene is really hard IMO, I'd prefer we avoid the macro here

Reply:
We will address this as part of the clean up effort. This method was chosen because `@btime` was much faster than the function option (which was needed for operations at every timestep), but we did realise that this poses= some restrictions on the design, from the macro hygiene POV. So far our tests should catch problems with the macro hygiene but it is on the [agenda](https://github.com/CliMA/ClimaCoupler.jl/issues/92) to review this, along with fresh performance tests.

Reply:
`@btime` of what was much faster? and what was the wall clock time? Does a non-macro version exist?

Reply:
`@btime` of a unit test of the @calendar_callback. I don't have the function on hand, because this was done months ago and okayed by reviewers. Since this is not part of this PR, can we address this separately?

Reply:
Yes, that's fine